source/common/http/header_map_impl.cc

```diff
@@ -650,5 +658,46 @@ HeaderMapImplUtility::getAllHeaderMapImplInfo() {
   return ret;
 }
 
+absl::optional<absl::string_view>
+RequestHeaderMapImpl::getTraceContext(absl::string_view key) const {
+  ASSERT(validatedLowerCaseString(key));
+  auto result = const_cast<RequestHeaderMapImpl*>(this)->getExisting(key);
```

Comment:
The `const_cast` doesn't looks appropriate here. Why `getExisting` has to be non-const?

Reply:
@lizan This is the original design of `HeaderMapImpl`. `HeaderMapImpl` also uses `const_cast` when implementing the 
original `get` method.
```
HeaderMap::GetResult HeaderMapImpl::get(const LowerCaseString& key) const {
  return HeaderMap::GetResult(const_cast<HeaderMapImpl*>(this)->getExisting(key));
}
```

> Why getExisting has to be non-const?

There are two main reasons I see.

First, the `staticLookup` that `getExisting` depends on is non-const. The reason `staticLookup` is non-const is that the `inlineHeaders` that `staticLookup` depends on is non-const.

Second, during the execution of the `getExsiting` method, `lazy_map_` may be initialized to speed up the search. This requires changing the state of HeaderMapImpl, so getExsiting needs to be non-const.

Reply:
Fair enough, I think we want to make `lazy_map_` mutable, but that is out of scope of this PR.